server/src/actors/project.rs

```diff
@@ -46,7 +44,8 @@ pub struct Project {
     config: Arc<Config>,
     manager: Addr<ProjectCache>,
     state: Option<Arc<ProjectState>>,
-    state_channel: Option<Shared<oneshot::Receiver<Option<Arc<ProjectState>>>>>,
+    state_channel: Option<Shared<oneshot::Receiver<Arc<ProjectState>>>>,
+    retry_after: Instant,
```

Comment:
This threshold is saved as `Instant` right now, because this is the most robust way to do so (even if someone fiddles with the system clock). However, that means that this field would not be persistable. If we'd prefer that, I would propose we switch to either `std::time::SystemTime` or `chrono::DateTime`.